src/approaches/pg3_approach.py

```diff
@@ -475,7 +480,8 @@ def get_next_state(atoms: _S, ground_nsrt: _A) -> _S:
         )
 
         def policy(atoms: _S) -> Optional[_A]:
-            return utils.query_ldl(ldl, set(atoms), goal)
+            objects = {o for a in atoms | goal for o in a.objects}
```

Comment:
this looks like the same dangerous pattern that we want to fix right? can we use the already-defined `objects` variable instead? (so just remove this line). hopefully this doesn't diminish the results you were seeing in painting with box-only, let's check after we merge this